wallet/src/main/scala/org/bitcoins/wallet/internal/AddressHandling.scala

```diff
@@ -93,67 +89,9 @@ private[wallet] trait AddressHandling extends WalletLogger {
       account: AccountDb,
       chainType: HDChainType
   ): Future[BitcoinAddress] = {
-    logger.debug(s"Getting new $chainType adddress for ${account.hdAccount}")
-
-    val lastAddrOptF = chainType match {
-      case HDChainType.External =>
-        addressDAO.findMostRecentExternal(account.hdAccount)
-      case HDChainType.Change =>
-        addressDAO.findMostRecentChange(account.hdAccount)
-    }
-
-    lastAddrOptF.flatMap { lastAddrOpt =>
-      val addrPath: HDPath = lastAddrOpt match {
-        case Some(addr) =>
-          val next = addr.path.next
-          logger.debug(
-            s"Found previous address at path=${addr.path}, next=$next")
-          next
-        case None =>
-          val chain = account.hdAccount.toChain(chainType)
-          val address = HDAddress(chain, 0)
-          val path = address.toPath
-          logger.debug(s"Did not find previous address, next=$path")
-          path
-      }
-
-      val addressDb = {
-        val pathDiff =
-          account.hdAccount.diff(addrPath) match {
-            case Some(value) => value
-            case None =>
-              throw new RuntimeException(
-                s"Could not diff ${account.hdAccount} and $addrPath")
-          }
-
-        val pubkey = account.xpub.deriveChildPubKey(pathDiff) match {
-          case Failure(exception) => throw exception
-          case Success(value)     => value.key
-        }
-
-        addrPath match {
-          case segwitPath: SegWitHDPath =>
-            AddressDbHelper
-              .getSegwitAddress(pubkey, segwitPath, networkParameters)
-          case legacyPath: LegacyHDPath =>
-            AddressDbHelper.getLegacyAddress(pubkey,
-                                             legacyPath,
-                                             networkParameters)
-          case nestedPath: NestedSegWitHDPath =>
-            AddressDbHelper.getNestedSegwitAddress(pubkey,
-                                                   nestedPath,
-                                                   networkParameters)
-        }
-      }
-      logger.debug(s"Writing $addressDb to DB")
-      val writeF = addressDAO.create(addressDb)
-      writeF.foreach { written =>
-        logger.debug(
-          s"Got ${chainType} address ${written.address} at key path ${written.path} with pubkey ${written.ecPublicKey}")
-      }
-
-      writeF.map(_.address)
-    }
+    val p = Promise[BitcoinAddress]
+    addressRequestQueue.add((account, chainType, p))
```

Comment:
It needs a doc describing that `java.lang.IllegalStateException` can be thrown when the queue is full, and also a unit test that checks it.